pkg/kubefedctl/federate/federate.go

```diff
@@ -76,20 +76,22 @@ var (
 
 type federateResource struct {
 	options.GlobalSubcommandOptions
-	typeName          string
-	resourceName      string
-	resourceNamespace string
-	output            string
-	outputYAML        bool
-	enableType        bool
-	federateContents  bool
+	typeName             string
+	resourceName         string
+	resourceNamespace    string
+	output               string
+	outputYAML           bool
+	enableType           bool
+	federateContents     bool
+	skipAPIResourceNames string
 }
 
 func (j *federateResource) Bind(flags *pflag.FlagSet) {
 	flags.StringVarP(&j.resourceNamespace, "namespace", "n", "default", "The namespace of the resource to federate.")
 	flags.StringVarP(&j.output, "output", "o", "", "If provided, the resource that would be created in the API by the command is instead output to stdout in the provided format.  Valid format is ['yaml'].")
 	flags.BoolVarP(&j.enableType, "enable-type", "e", false, "If true, attempt to enable federation of the API type of the resource before creating the federated resource.")
 	flags.BoolVarP(&j.federateContents, "contents", "c", false, "Applicable only to namespaces. If provided, the command will federate all resources within the namespace after federating the namespace.")
+	flags.StringVarP(&j.skipAPIResourceNames, "skip-api-resources", "s", "", "Comma separated names of the api resources to skip when federating contents in a namespace.")
```

Comment:
I think a more detailed help text is suggested here, since the 'name' of the api resource could be either a group or a group-qualified plural name.  Examples for each are suggested.

Also, I think `flags.StringSliceP` is probably a better choice.

Reply:
+1 for `flags.StringSliceP`

Reply:
Updated the help text.